spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -0,0 +1,435 @@
+    DataConverter.setAttributeValue(context, serializerFlagBuilder, true,
+      scala.reflect.runtime.universe.typeOf[Boolean])
+    moduleBuilder.putAttr("is_labor_module", serializerFlagBuilder.build)
+  }
+}
+
+private[bigdl] object KerasLayer {
+    def fuse[T: ClassTag](sLayer: AbstractModule[Activity, Activity, T],
+        activation: TensorModule[T],
+        inputShape: Shape)
+        (implicit ev: TensorNumeric[T]): AbstractModule[Activity, Activity, T] = {
+      if (activation == null) {
+        return sLayer
+      }
+      val seq = KSequential[T]()
+      seq.add(InputLayer(inputShape = KerasLayer.removeBatch(inputShape)))
+      seq.add(sLayer)
+      seq.add(activation)
+      seq.setName(sLayer.getName())
+      return seq
+    }
+
+   def addBatch(shape: Shape): Shape = {
+    if (shape == null) {
+      return null
+    }
+    if (shape.isInstanceOf[SingleShape]) {
+      Shape((List(-1) ++ shape.toSingle()).toArray)
+    } else {
+      Shape(shape.toMulti().map {addBatch(_)})
+    }
+  }
+
+  def removeBatch(shape: Shape): Shape = {
+    if (shape == null) {
+      return null
+    }
+    if (shape.isInstanceOf[SingleShape]) {
+      Shape((shape.toSingle().slice(1, shape.toSingle().length)).toArray)
+    } else {
+      Shape(shape.toMulti().map {addBatch(_)})
+    }
+  }
+}
+
+/**
+ * KerasModule is the basic component of all Keras-like Layer.
+ * It forward activities and backward gradients, and can be mixed with other AbstractMoudule.
+ *
+ * @tparam A Input data type
+ * @tparam B Output data type
+ * @tparam T Numeric type of parameter(e.g. weight, bias). Only support float/double now
+ * @param mInputShape inputshape for a layer which is just a shape of a record without batch.
+ */
+abstract class KerasLayer[A <: Activity: ClassTag, B <: Activity: ClassTag, T: ClassTag]
+(mInputShape: Shape = null)(implicit ev: TensorNumeric[T]) extends AbstractModule[A, B, T]{
+
+  private val batchInputShape = KerasLayer.addBatch(mInputShape)
+
+  var labor: AbstractModule[A, B, T] = null
+
+  override def output: B = labor._output
+
+  override def gradInput: A = labor._gradInput
+
+  override def inputShapeValue: Shape = labor._inputShapeValue
+
+  override def outputShapeValue: ArrayBuffer[Shape] = labor._outputShapeValue
+
+  // scalastyle:off
+  override def output_= (value: B): Unit = labor._output = value
+
+  override def gradInput_=(value: A): Unit = {
+    labor._gradInput = value
+  }
+
+  override def inputShapeValue_=(value: Shape): Unit = {
+    labor._inputShapeValue = value
+  }
+
+  override def outputShapeValue_=(value: ArrayBuffer[Shape]): Unit = {
+   labor._outputShapeValue = value
+  }
+  // scalastyle:on
+
+  override def compatibleWithKeras(): Boolean = true
+
+  override def compatibleWithTorch(): Boolean = false
+
+  override def getInputShape(): Shape = {
+    if (batchInputShape != null) {
+      batchInputShape
+    } else if (this.labor == null) {
+      null
```

Comment:
I don't think return null is a good practice.

Reply:
tons of the code within bigdl return null for now. i.e `    if (node.element.isInstanceOf[WithoutInput]) return null
`

Reply:
What does it mean when shape is null?